File: LeetCode-2023/2319.判断矩阵是否是一个-x-矩阵.py
#
# @lc app=leetcode.cn id=2319 lang=python3
#
# [2319] 判断矩阵是否是一个 X 矩阵
#

# @lc code=start
class Solution:
    def checkXMatrix(self, grid: List[List[int]]) -> bool:
        n = len(grid)
        for i in range(0, n):
            for j in range(0, n):
                if i == j or i + j == n - 1:
                    if grid[i][j] == 0:
                        return False
                else:
                    if grid[i][j] != 0:
                        return False
        return True
# @lc code=end

File: LeetCode-2023/2319.判断矩阵是否是一个-x-矩阵.java
/*
 * @lc app=leetcode.cn id=2319 lang=java
 *
 * [2319] 判断矩阵是否是一个 X 矩阵
 */

// @lc code=start
class Solution {
    public boolean checkXMatrix(int[][] grid) {
        int n = grid.length;
        for(int i = 0; i < n; i++) {
            for(int j = 0; j < n; j++) {
                if(i == j || i + j == n - 1) {
                    if(grid[i][j] == 0) {
                        return false;
                    }
                } else {
                    if(grid[i][j] != 0) {
                        return false;
                    }
                }
            }
        }
        return true;
    }
}
// @lc code=end

